File: src/main/java/no/fintlabs/mapping/DokumentObjektMappingService.java
package no.fintlabs.mapping;

import no.fint.model.resource.Link;
import no.fint.model.resource.arkiv.noark.DokumentobjektResource;
import no.fintlabs.model.instance.DokumentobjektDto;
import org.springframework.stereotype.Service;

import java.util.Collection;
import java.util.List;
import java.util.Map;
import java.util.UUID;

@Service
public class DokumentObjektMappingService {

    public List<DokumentobjektResource> toDokumentobjektResource(
            Collection<DokumentobjektDto> dokumentobjektDto,
            Map<UUID, Link> fileArchiveLinkPerFileId
    ) {
        return dokumentobjektDto
                .stream()
                .map(doDto -> toDokumentobjektResource(doDto, fileArchiveLinkPerFileId))
                .toList();
    }

    public DokumentobjektResource toDokumentobjektResource(
            DokumentobjektDto dokumentobjektDto,
            Map<UUID, Link> fileArchiveLinkPerFileId
    ) {
        DokumentobjektResource dokumentobjektResource = new DokumentobjektResource();
        dokumentobjektResource.addVariantFormat(dokumentobjektDto.getVariantformat());
        dokumentobjektResource.addFilformat(dokumentobjektDto.getFilformat());
        dokumentobjektResource.addReferanseDokumentfil(
                fileArchiveLinkPerFileId.get(dokumentobjektDto.getFileReference().getFileId())
        );
        return dokumentobjektResource;
    }

}

File: src/main/kotlin/no/novari/flyt/archive/gateway/dispatch/mapping/DokumentObjektMappingService.kt
package no.novari.flyt.archive.gateway.dispatch.mapping

import no.novari.fint.model.resource.Link
import no.novari.fint.model.resource.arkiv.noark.DokumentobjektResource
import no.novari.flyt.archive.gateway.dispatch.model.instance.DokumentobjektDto
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class DokumentObjektMappingService {
    fun toDokumentobjektResource(
        dokumentobjektDto: Collection<DokumentobjektDto>,
        fileArchiveLinkPerFileId: Map<UUID, Link>,
    ): List<DokumentobjektResource> = dokumentobjektDto.map { toDokumentobjektResource(it, fileArchiveLinkPerFileId) }

    fun toDokumentobjektResource(
        dokumentobjektDto: DokumentobjektDto,
        fileArchiveLinkPerFileId: Map<UUID, Link>,
    ): DokumentobjektResource =
        DokumentobjektResource().apply {
            dokumentobjektDto.variantformat?.let(Link::with)?.let(::addVariantFormat)
            dokumentobjektDto.filformat?.let(Link::with)?.let(::addFilformat)
            dokumentobjektDto.fileId?.let(fileArchiveLinkPerFileId::get)?.let(::addReferanseDokumentfil)
        }
}
